api/v1alpha1/pod_container_types.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2020 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import v1 "k8s.io/api/core/v1"
+
+// +kubebuilder:object:generate=true
+
+// PodImage represents the image information for a container that is used
+// to build the StatefulSet.
+type PodImage struct {
+	// (Required) Container image with supported CockroachDB version
+	Name string `json:"name,omitempty"`
```

Comment:
More of a musing than a comment: Would it be worth while to split `Name` into image and tag? Image could default to `cockroachdb/cockroach`.

Reply:
What is the value 🤷‍♂️

Reply:
Also, how do we support digests then?  For example:

```
docker pull frolvlad/alpine-miniconda3:python3.7@sha256:9bc9c096713a6e47ca1b4a0d354ea3f2a1f67669c9a2456352d28481a6ce2fbe
```

There are multiple formats for "tagging", and I would rather just have the user provide the entire string.  Easy :)

Reply:
Ah, good point! I didn't think of that.